libs/database/src/lib/services/ranker.service.ts

```diff
@@ -118,114 +117,80 @@ export class RankerService {
     excludeContents?: any[]
   ) => {
     console.log('exclude', excludeContents);
-    let prefix_feeds_payload: FeedItemPayloadItem[] = [];
-    let prefix_feeds: DefaultContent[];
-    //if no pagination = add default
+
+    let defaultFeeds: DefaultContent[];
+
     if (!(query.untilId || query.sinceId)) {
-      prefix_feeds = await this._defaultContentModel
+      defaultFeeds = await this._defaultContentModel
         .find({ index: { $gte: 0 } })
         .populate('content')
         .sort({ index: 1 });
-      prefix_feeds_payload = prefix_feeds.map(
-        (item) =>
-          ({
-            id: 'default',
-            feature: {
-              slug: 'feed',
-              key: 'feature.feed',
-              name: 'Feed',
-            },
-            circle: {
-              id: 'for-you',
-              key: 'circle.forYou',
-              name: 'For You',
-              slug: 'forYou',
-            },
-            payload: toSignedContentPayloadItem(item.content),
-            type: 'content',
-          } as FeedItemPayloadItem)
-      );
     }
-    const filter = createCastcleFilter(
+    const filters = createCastcleFilter(
       {
         countryCode: viewer.geolocation?.countryCode?.toLowerCase() ?? 'en',
         content: { $nin: excludeContents },
       },
       { ...query, sinceId: query.untilId, untilId: query.sinceId }
     );
-    const feedItems = await this._guestFeedItemModel
-      .find(filter)
+
+    const guestFeeds = await this._guestFeedItemModel
+      .find(filters)
       .populate('content')
       .limit(query.maxResults)
       .sort({ score: -1, createdAt: -1 })
       .exec();
 
-    let authors = feedItems.map((feedItem) => feedItem.content.author);
-    authors = authors.concat(
-      feedItems
-        .filter((feedItem) => feedItem.content.originalPost)
-        .map((feedItem) => feedItem.content.originalPost.author)
+    const feeds = [...defaultFeeds, ...guestFeeds];
```

Comment:
ลองใช้ aggregation -> facet: defaultFeeds, guestFeeds -> facet lookup users/relationships/engagements (in defaultFeed.author+guestFeed.author)